src/fr/lugnicascans/src/eu/kanade/tachiyomi/extension/fr/lugnicascans/LugnicaScans.kt

```diff
@@ -0,0 +1,125 @@
+package eu.kanade.tachiyomi.extension.fr.lugnicascans
+
+import eu.kanade.tachiyomi.network.GET
+import eu.kanade.tachiyomi.source.model.FilterList
+import eu.kanade.tachiyomi.source.model.MangasPage
+import eu.kanade.tachiyomi.source.model.Page
+import eu.kanade.tachiyomi.source.model.SChapter
+import eu.kanade.tachiyomi.source.model.SManga
+import eu.kanade.tachiyomi.source.online.HttpSource
+import keiyoushi.utils.parseAs
+import okhttp3.Request
+import okhttp3.Response
+
+class LugnicaScans : HttpSource() {
+    override val name = "Lugnica Scans"
+
+    override val baseUrl = "https://lugnica-scans.com"
+
+    override val lang = "fr"
+
+    override val supportsLatest = true
+
+    override val client = super.client.newBuilder()
+        .addInterceptor { chain ->
+            val request = chain.request()
+            if (request.url.fragment != "manga") return@addInterceptor chain.proceed(request)
+
+            val slug = request.url.pathSegments.last()
+            val newUrl = "$baseUrl/api/get/card/$slug"
+
+            val newRequest = request.newBuilder().url(newUrl).build()
+
+            return@addInterceptor chain.proceed(newRequest)
+        }.addInterceptor { chain ->
+            val request = chain.request()
+            if (request.url.fragment != "chapter") return@addInterceptor chain.proceed(request)
+
+            val slug = request.url.pathSegments.takeLast(2).joinToString("/")
+            val newUrl = "$baseUrl/api/get/chapter/$slug"
+
+            val newRequest = request.newBuilder().url(newUrl).build()
+
+            return@addInterceptor chain.proceed(newRequest)
+        }.build()
+
+    // Popular
+    override fun popularMangaRequest(page: Int): Request = throw UnsupportedOperationException()
+    override fun popularMangaParse(response: Response): MangasPage = throw UnsupportedOperationException()
+
+    // Latest
+    override fun latestUpdatesRequest(page: Int): Request = GET("$baseUrl/api/get/homegrid/$page", headers)
+
+    override fun latestUpdatesParse(response: Response): MangasPage {
+        val rawMangas = response.parseAs<List<HomePageManga>>()
+
+        val mangas = rawMangas.map { manga ->
+            SManga.create().apply {
+                title = manga.manga_title
+                setUrlWithoutDomain("/manga/${manga.manga_slug}#manga")
+
+                thumbnail_url = "$baseUrl/upload/min_cover/${manga.manga_image}"
+            }
+        }
+
+        return MangasPage(mangas, mangas.size == 10)
+    }
+
+    // Search
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request = throw UnsupportedOperationException()
+
+    override fun searchMangaParse(response: Response): MangasPage = throw UnsupportedOperationException()
```

Comment:
is there any way to get all manga on website to implement client side search? otherwise throw exception with a message.

Reply:
This should be doable but could you explain me how to access mangas in the reader's database ? Or should I fetch the "popular" section ? (There are 41 mangas)